lib/filterx/object-datetime.c

```diff
@@ -200,6 +200,64 @@ _repr(FilterXObject *s, GString *repr)
   return TRUE;
 }
 
+FilterXObject *
+filterx_datetime_strptime(GPtrArray *args)
+{
+  if (args == NULL || args->len < 2)
+    {
+      msg_error("FilterX: Failed to create datetime object: invalid number of arguments. "
+                "Usage: strptime(time_str, format_str0, ..., format_strN)");
+      return NULL;
+    }
+
+  FilterXObject *object = g_ptr_array_index(args, 0);
+  if (!object || !filterx_object_is_type(object, &FILTERX_TYPE_NAME(string)))
+    {
+      msg_error("FilterX: Failed to create datetime object: bad argument. "
+                "Usage: strptime(time_str, format_str0, ..., format_strN)",
+                evt_tag_int("arg_pos", 0));
+      return NULL;
+    }
+  const gchar *timestr = filterx_string_get_value(object, NULL);
+
+  FilterXObject *result = NULL;
+
+  WallClockTime wct = WALL_CLOCK_TIME_INIT;
+  UnixTime ut = UNIX_TIME_INIT;
+  gchar *end = NULL;
+
+  for (int i = 1; i < args->len; i++)
+    {
+      FilterXObject *time_fmt_obj = g_ptr_array_index(args, i);
+      if (!time_fmt_obj || !filterx_object_is_type(time_fmt_obj, &FILTERX_TYPE_NAME(string)))
+        {
+          msg_error("FilterX: Failed to create datetime object: bad argument. "
+                    "Usage: strptime(time_str, format_str0, ..., format_strN)",
+                    evt_tag_int("arg_pos", i));
+          return NULL;
+        }
+
+      const gchar *time_fmt = filterx_string_get_value(time_fmt_obj, NULL);
+      end = wall_clock_time_strptime(&wct, time_fmt, timestr);
+      if (!end)
+        {
+          msg_error("filterx: unable to parse time",
+                    evt_tag_str("time_string", timestr),
+                    evt_tag_str("format", time_fmt));
```

Comment:
I think it is acceptable for some formats to not match. I think we should not handle this as an error, and we should just continue with the next format. Maybe we can keep this message on trace or debug level.

Reply:
I think you mixed up the messages:
* "FilterX: Failed to create datetime object: bad argument." should be error
* "filterx: unable to parse time" should be debug

Reply:
true, nice catch, thanks